app/_src/kubernetes-ingress-controller/guides/migrate/kongingress.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Removing KongIngress
+---
+
+## konghq.com/headers.* annotation
+
+The [`konghq.com/headers.*` annotation][headers-annotation] uses a special
+format to set headers. The string after the `.` in the annotation name is the
+header name and the annotation value is the header value. For example, to apply
+`x-custom-header-a: example,otherexample` and `x-custom-header-b: example`
+headers to requests, KongIngress configuration would look like:
+
+```yaml
+route:
+  headers:
+    x-custom-header-a:
+    - example
+    - otherexample
+    x-custom-header-b:
+    - example
+```
+
+The equivalent annotation configuration looks like:
+
+```text
+konghq.com/headers.x-custom-header-a: example,otherexample
+konghq.com/headers.x-custom-header-b: example
+```
+
+Note that you cannot apply multiple instances of the same header annotation to
+set multiple header values: you must set the CSV format within a single header.
+
+## KongUpstreamPolicy
+
+The `upstream` section of KongIngress resources contains a complex object that does not easily fit in annotations. 3.x uses the new KongUpstreamPolicy resource to configure upstream settings. The `spec` field of KongUpstreamPolicy is the same as the `upstream` section of KongIngress. For example, if you previously used a KongIngress like:
```

Comment:
```suggestion
The `upstream` section of KongIngress resources contains a complex object that does not easily fit in annotations. Version 3.x uses the new KongUpstreamPolicy resource to configure upstream settings. The `spec` field of `KongUpstreamPolicy` is the same as the `upstream` section of KongIngress. For example, if you previously used a `KongIngress`:
```

Reply:
Shouldn't we use the code syntax for all the CRD name references?

```suggestion
The `upstream` section of `KongIngress` resources contains a complex object that does not easily fit in annotations. 3.x uses the new `KongUpstreamPolicy` resource to configure upstream settings. The `spec` field of `KongUpstreamPolicy` is the same as the `upstream` section of `KongIngress`. For example, if you previously used a `KongIngress` like:
```